### PR TITLE
feat(radio,cpn): Startup splash screen and sound changes.

### LINKED ISSUE #391: Ability to enable/disable startup sound
I've always thought it was rather silly you often have the option of enabling or disabling the startup splash screen, but not the startup sound, so have to resort to deleting it if you don't want it to play. I'm clearly not the only one thinking this ;)

> first thing I do when I move to a new sd card is delete the startup sound file. Instead of having to do this, I think adding an enable check box would be a nice addition.

I would envisage this as a checkbox in the radio settings...

https://www.rcgroups.com/forums/showpost.php?p=47392067&postcount=110

### LINKED ISSUE #2537: Add delay and sound to splash screen image
### Is there an existing issue for this feature request?

- [X] I have searched the existing issues

### Describe the solution you'd like

Please add feature adding delay on both image and  to add sound to splash screen. 

### Additional context

_No response_

### LINKED ISSUE #3438: Splash screen and welcome audio
### Is there an existing issue for this feature request?

- [X] I have searched the existing issues

### Is your feature request related to a problem?

No

### Describe the solution you'd like

Is it possible to have the welcome splash screen and welcome audio come on at the same time and last for 7 seconds. Similar to OpenTX. Thanks

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by philmoz**
Fixes #3438, fixes #2537, fixes #391 

Summary of changes:
- Enable startup splash screen duration for color LCD radios.
- Add option to disable startup sound on all radios.
- Add Companion support for above changes.
- Move mixer task startup to after splash screen is done on color LCD radios so that the startup sound is played first.

Notes:
- The startup splash screen cannot be disabled for color LCD radios. The minimum setting is 1 second. This is because the radio.yml file has not been loaded when the splash screen is displayed so the property to disable it is unknown.
- There is an initial delay on color LCD radios before the startup sound is played. This is due to the initialisation of the theme and loading the radio & model files from the SD card. The length of the delay will depend on the radio, SD card speed, theme background image and number of model files, and this delay will determine the actual minimum time for the splash screen to be shown.
- Moving the mixer task startup does not seem to have caused any adverse side effects on my QX7S, TX16S Mk2 or the simulator. It probably needs more testing on other radios though.

Screen Shot 2023-04-08 at 8 13 12 pm

Screen Shot 2023-04-08 at 8 13 55 pm

Screen Shot 2023-04-08 at 8 14 20 pm

### CODE REVIEW

**Comment by pfeerick** [comment]
Hm, be interesting to see if this fixes #135 also

**Comment by philmoz** [comment]
> Hm, be interesting to see if this fixes #135 also

I think it should fix most cases; but it's a hard one to test.

I tested a simple widget and calling playSound in the refresh method is ok, the sound plays after the welcome sound.
Calling playSound from the create method will, however, still play before the welcome sound because Lua scripts are initialised first.

**Comment by pfeerick** [comment]
It's should actually to test this particular case - the yappu telemetry widget plays a "yappu telemetry ready" sound as it starts, so if that no longer precedes the ETX `hello.wav` when the active model on power on has the widget installed, it's fixed. I guess it will depend on where the yappu playsound is.

**Comment by philmoz** [comment]
> I guess it will depend on where the yappu playsound is.

Yaapu plays its startup sound in the create function so this change does not specifically fix #135 (pfeerick: added "not" as clear contradiction of "Calling playSound from the create method will, however, still play before the welcome sound because Lua scripts are initialised first.")

**Comment by ulfhedlund** [comment]
Is there a string for translation of "Play Startup Sound" in Companion?

**Comment by pfeerick** [comment]
Hi all... hope you've been enjoying your weekend :) Can I get a translation for this one please. Screenshots in the initial post for context. 

```c
#define TR_PLAY_HELLO "Startup Sound"
```

- [ ] @zyren CN / TW
- [ ] @Eldenroot CZ
- [ ] @HThuren DA
- [ ] @ParkerEde / @TheIsotopes DE
- [ ] @Pat6874 FR
- [ ] @offer-shmuely HE
- [ ] @robustini IT
- [ ] @ToshihiroMakuuchi JP
- [ ] @ulfhedlund SE
- [ ] @ajjjjjjjj PL
- [ ] @zandorsp PT

**Comment by Pat6874** [comment]
FR:

#define TR_PLAY_HELLO "Son de démarrage"


Le dim. 16 juil. 2023, 07:01, Peter Feerick ***@***.***> a
écrit :

**Comment by ulfhedlund** [comment]
SE:#define TR_PLAY_HELLO "Startljud"Den 16 juli 2023 07:01 skrev Peter Feerick ***@***.***>:
Hi all... hope you've been enjoying your weekend :) Can I get a translation for this one please. Screenshots in the initial post for context.
#define TR_PLAY_HELLO "Startup Sound"
 @zyren CN / TW @Eldenroot CZ @HThuren DA @ParkerEde / @TheIsotopes DE @Pat6874 FR @offer-shmuely HE @robustini IT @ToshihiroMakuuchi JP @ulfhedlund SE @ajjjjjjjj PL @zandorsp PT

**Comment by offer-shmuely** [comment]
#define TR_PLAY_HELLO "צליל אתחול"

**Comment by froqstar** [comment]
DE:
#define TR_PLAY_HELLO "Startton"

**Comment by HThuren** [comment]
DA
#define TR_PLAY_HELLO "Startop lyd"

... and nice weekend to you too

**Comment by zyren** [comment]
CN
```c
#define TR_PLAY_HELLO "开机语音"
```

TW
```c
#define TR_PLAY_HELLO "開機語音"
```

**Comment by Eldenroot** [comment]
CZ

#define TR_PLAY_HELLO "Zvuk při spuštění"

**Comment by zandorsp** [comment]
PT
 ```c
#define TR_PLAY_HELLO "Som ao ligar"
```

**Comment by robustini** [comment]
IT
#define TR_PLAY_HELLO "Suono all'accensione"

**Comment by philmoz** [comment]
> FR: #define TR_PLAY_HELLO "Son de démarrage"

> CZ: #define TR_PLAY_HELLO "Zvuk při spuštění"

> IT: #define TR_PLAY_HELLO "Suono all'accensione"

I think these are too long for small B&W LCD screens, can we have abbreviated versions please.

**Comment by philmoz** [comment]
To make more room for the setup labels, the controls could be right justified on small B&W LCD screen.
Is this worth doing?

Example:
Screenshot 2023-07-17 at 8 50 38 am

Current layout:
Screenshot 2023-07-17 at 8 54 12 am

**Comment by ToshihiroMakuuchi** [comment]
JA
#define TR_PLAY_HELLO "起動時サウンド"

**Comment by robustini** [comment]
IT
#define TR_PLAY_HELLO TR("Suono accens.", "Suono all'accensione")

**Comment by froqstar** [comment]
I forgot about the unlimited possibilities of the color LCD :)

DE
#define TR_PLAY_HELLO TR("Startton", "Startton abspielen")

**Comment by ajjjjjjjj** [comment]
PL
#define TR_PLAY_HELLO "Dźwięk uruchomienia"

**Comment by Pat6874** [comment]
I think, it s better right justified
Patrick 

Le 17 juil. 2023, 00:55, à 00:55, philmoz ***@***.***> a écrit:

**Comment by Eldenroot** [comment]
> > FR: #define TR_PLAY_HELLO "Son de démarrage"
> 
> > CZ: #define TR_PLAY_HELLO "Zvuk při spuštění"
> 
> > IT: #define TR_PLAY_HELLO "Suono all'accensione"
> 
> I think these are too long for small B&W LCD screens, can we have abbreviated versions please.

FOR CZ (BW radios, shortened version of text)

"Zvuk spuštění"

**Comment by Pat6874** [comment]
FR: #define TR_PLAY_HELLO "Son accueil"

Le mer. 19 juil. 2023 à 22:18, Eldenroot ***@***.***> a
écrit :

**Comment by pfeerick** [comment]
Sorry for the delay in getting around to this.

1. This is fantastic... setting it to the shortest delay my TX16S starts up faster (warnings permitting). 
2. Companion side seems fine. 
3. B&W 128 seems fine, right justified setup menu works ok

It doesn't actually fix #135 as we thought back at https://github.com/EdgeTX/edgetx/pull/3451#issuecomment-1501017132 - it still manages to sneak in before "welcome to edgetx" (or if it did before, something else has changed). That's not much of an issue, it was never specifically targetting that.

The only issue (and it's a doozie...) set the startup time to 8s (or longer), and restart the radio 🤭 My TX16S at least doesn't like that...

**Comment by philmoz** [comment]
> The only issue (and it's a doozie...) set the startup time to 8s (or longer), and restart the radio 🤭 My TX16S at least doesn't like that...

Ooops. Any ideas?

**Comment by philmoz** [comment]
Suspending the watchdog timer seems to fix it - hopefully this is the correct way to do it.

**Comment by pfeerick** [comment]
Interesting... it fitted the timeline of the watchdog, but I had thought RTOS_WAIT_TICKS would have satisfied it... :shrug: 

@raphaelcoeffic @gagarinlg Your thoughts?

File: radio/src/opentx.h [hunk 1]
```diff
@@ -340,9 +340,7 @@ inline bool SPLASH_NEEDED()
 #endif
 }
 
-#if defined(PCBHORUS)
-  #define SPLASH_TIMEOUT               0 /* we use the splash duration to load stuff from the SD */
-#elif defined(PCBTARANIS)
+#if defined(PCBHORUS) || defined(PCBTARANIS) || defined(PCBNV14)
```
Comment (pfeerick) [inline radio/src/opentx.h:343]: This was the only other oddity... normally when PCBHORUS and PCBNV14 are together, they can be replaced with COLORLCD, making it more generic, and PCBTARANIS is every single B&W radio ... so what is this really conditional on now, when would ELSE ever trigger? Perhaps it only needs to be conditional on SPLASH?

Comment (3djc) [reply radio/src/opentx.h:343]: Historically all arm 9x and earlier mega radio

Comment (pfeerick) [reply radio/src/opentx.h:343]: Thanks, ok, so the else can indeed go as we've dropped those radios, and I guess `#if defined(SPLASH)` will do then.

**Comment by philmoz** [comment]
> Do not stop the watchdog!

watchdogSuspend() is used elsewhere - is it not safe to use while the splash screen is being shown?

**Comment by gagarinlg** [comment]
> > Do not stop the watchdog!
> 
> watchdogSuspend() is used elsewhere - is it not safe to use while the splash screen is being shown?

Then we should redesign those parts.
I would prefer to regularly wake up, kick the watchdog and the go back to sleep, when the timeout is not reached

**Comment by philmoz** [comment]
I've changed the splash timeout loop to use WDG_RESET instead.

**Pull request merged**